modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala

```diff
@@ -57,191 +54,151 @@ class InstanceRoutes(
+          pathPrefix(Segment) { domain =>
+            val domainId = DomainId(orgId, domain)
+            (pathEndOrSingleSlash & get & searchQueryParams) { (pagination, filterOpt, termOpt, deprecatedOpt) =>
+                traceName("listInstancesOfDomain") {
+                  val filter = filterFrom(deprecatedOpt, filterOpt, querySettings.nexusVocBase)
+                  instanceQueries.list(domainId, filter, pagination, termOpt) buildResponse(base, pagination)
                 }
-              }
             } ~
-            pathPrefix(Segment) { versionString =>
-              Version(versionString) match {
-                case None          =>
-                  exceptionHandler(CommandRejected(SchemaRejection.IllegalVersionFormat))
-                case Some(version) =>
-                  val schemaId = SchemaId(DomainId(orgId, domain), name, version)
-
-                  pathEndOrSingleSlash {
-                    (get & paginatedAndFiltered) { (pagination, filterOpt) =>
-                      parameter('deprecated.as[Boolean].?) { deprecated =>
-                        traceName("listInstancesOfSchema") {
-                          val filter = Filter(deprecatedAndRev(deprecated)) and filterOpt.map(_.expr)
-                          buildResponse(instanceQueries.list(schemaId, filter, pagination), pagination)
-                        }
-                      }
-                    }
-                  } ~
-                  (pathEndOrSingleSlash & post) {
-                    entity(as[Json]) { json =>
-                      traceName("createInstance") {
-                        onSuccess(instances.create(schemaId, json)) { ref =>
-                          complete(StatusCodes.Created -> ref)
-                        }
-                      }
-                    }
-                  } ~
-                  pathPrefix(Segment) { id =>
-                    val instanceId = InstanceId(schemaId, id)
-                    (pathPrefix("outgoing") & pathEndOrSingleSlash & get) {
-                      parameter('deprecated.as[Boolean].?) { deprecated =>
-                        paginatedAndFiltered.apply { (pagination, filterOpt) =>
-                          val filter = Filter(deprecatedAndRev(deprecated)) and filterOpt.map(_.expr)
-                          buildResponse(instanceQueries.outgoing(instanceId, filter, pagination), pagination)
-                        }
+            pathPrefix(Segment) { name =>
+              (pathEndOrSingleSlash & get & searchQueryParams) { (pagination, filterOpt, termOpt, deprecatedOpt) =>
+                traceName("listInstancesOfSchemaName") {
+                  val filter = filterFrom(deprecatedOpt, filterOpt, querySettings.nexusVocBase)
+                  instanceQueries.list(domainId, name, filter, pagination, termOpt) buildResponse(base, pagination)
+                }
+              } ~
+              pathPrefix(Segment) { versionString =>
+                Version(versionString) match {
```

Comment:
Not necessarily to be addressed in this change, but we should have a version directive with its rejection and handling to simplify routes that involve a version.

Reply:
Done